frontend/js/src/common/Username.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from "react";
+import { Link } from "react-router-dom";
+import useUserFlairs from "../utils/FlairLoader";
+
+function Username({
+  username,
+  elementType,
+  hideFlair = false,
+  showLink = false,
+  ...otherProps
+}: {
+  username: string;
+  elementType?: keyof JSX.IntrinsicElements;
+  hideFlair?: boolean;
+  showLink?: boolean;
+  [key: string]: any;
+}) {
+  const Element = elementType;
+  const flairType = useUserFlairs(username);
+
+  if (showLink) {
+    return (
+      <Link
+        to={`/user/${username}/`}
+        {...otherProps}
+        className={`${otherProps?.className || ""} ${
+          !hideFlair ? flairType || "" : ""
```

Comment:
I made it so you always need to apply the class `flair` as well as the actual flair type, that seemed a more safe option to ensure we don't end up with a wacky css animation somewhere not intended...
See #3027 

That being said, "flair" need to be added here along with flairType/